cs-office-365/src/main/java/io/cloudslang/content/office365/actions/utils/GetAuthorizationToken.java

```diff
@@ -0,0 +1,130 @@
+import static io.cloudslang.content.office365.utils.Outputs.AuthorizationOutputs.AUTH_TOKEN_TYPE;
+import static io.cloudslang.content.utils.OutputUtilities.getFailureResultsMap;
+import static io.cloudslang.content.utils.OutputUtilities.getSuccessResultsMap;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+public class GetAuthorizationToken {
+
+    /**
+     * @param loginType      Login method according to application type
+     *                       Valid values: 'API', 'Native'
+     *                       Default: 'API'
+     * @param clientId       Service Client ID
+     * @param clientSecret   Service Client Secret
+     * @param username       Office 365 username
+     * @param password       Office 365 password
+     * @param loginAuthority The authority URL. Usually, the format for this input is:
+     *                       'https://login.windows.net/TENANT_NAME/oauth2/token' where TENANT_NAME is your application
+     *                       tenant.
+     * @param resource       The resource URL
+     *                       Default: 'https://graph.microsoft.com'
+     * @param proxyHost      Proxy server used to access the web site
+     * @param proxyPort      Proxy server port
+     *                       Default: '8080'
+     * @param proxyUsername  User name used when connecting to the proxy
+     * @param proxyPassword  The proxy server password associated with the proxyUsername input value
+     * @return The authorization token for Office 365
+     */
+    @Action(name = "Get the authorization token for Office 365",
+            outputs = {
+                    @Output(value = RETURN_RESULT, description = RETURN_RESULT_DESC),
+                    @Output(value = RETURN_CODE, description = RETURN_CODE_DESC),
+                    @Output(value = AUTH_TOKEN_TYPE, description = AUTH_TOKEN_TYPE_DESC),
+                    @Output(value = EXCEPTION, description = EXCEPTION_DESC)
+            },
+            responses = {
+                    @Response(text = SUCCESS, field = RETURN_CODE, value = ReturnCodes.SUCCESS, matchType = COMPARE_EQUAL, responseType = RESOLVED, description = SUCCESS_DESC),
+                    @Response(text = FAILURE, field = RETURN_CODE, value = ReturnCodes.FAILURE, matchType = COMPARE_EQUAL, responseType = ERROR, description = FAILURE_DESC)
+            })
+    public Map<String, String> execute(@Param(value = LOGIN_TYPE, description = LOGIN_TYPE_DESC) String loginType,
+                                       @Param(value = CLIENT_ID, required = true, description = CLIENT_ID_DESC) String clientId,
+                                       @Param(value = CLIENT_SECRET, encrypted = true, description = CLIENT_SECRET_DESC) String clientSecret,
+                                       @Param(value = USERNAME, description = USERNAME_DESC) String username,
```

Comment:
mark as final

Reply:
Unfortunately, the implementation would require me to add new inputs with different names. Good point, btw ;).